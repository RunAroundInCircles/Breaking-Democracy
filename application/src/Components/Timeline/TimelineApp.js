import React,{Component} from 'react';
import TimelineEvent from './TimelineEvent.js';
import Timeline from './Timeline.js';

/**
 * TimelineApp component of the app that renders and returns a Timeline which
 * gives the physical timeline object and a TimelineEvent
 * that is given all current events happening on the timeline.
 * @extends React
 */
class TimelineApp extends Component {
// This code snippet will be used later in production
/*    constructor(props) {
		super(props);
    }
    */
   
	/**
 	* Renders a Timeline and a TimelineEvent given all current events.
 	* @return returns the div of a Timeline and a TimelineEvent Component that has all current events passed in.
 	*/
    render() {
        return(
            <div className="timeline-app">
                <Timeline/>
                <TimelineEvent timelineEvents={this.props.timelineEvents}/>
            </div>
        )
    }
}

export default TimelineApp;
