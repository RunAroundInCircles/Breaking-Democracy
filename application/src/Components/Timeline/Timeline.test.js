import React from 'react';
import TimelineApp from './TimelineApp.js'
import timelineevents from './TimelineData.json';
import renderer from 'react-test-renderer';

/**
 * Checks to see if the Timeline is correctly rendered
 */
test('renders TimelineApp page', () => {
	const tree = renderer
	.create(<TimelineApp timelineevents={timelineevents}/>)
	expect(tree).toMatchSnapshot();
});
