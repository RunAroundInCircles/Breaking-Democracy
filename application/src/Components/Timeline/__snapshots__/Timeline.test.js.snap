// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TimelineApp page 1`] = `
<div
  className="timeline-app"
>
  <div
    className="timeline-list"
  >
    <h1
      className="Event-head"
    >
      Timeline
    </h1>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "80vh",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100vw",
        }
      }
    >
      <svg
        height="50"
        width="50"
      >
        <polygon
          points="0,25 50,0 50,50"
          style={
            Object {
              "fill": "grey",
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
        />
      </svg>
      <svg
        height="50"
        width="100vw"
      >
        <line
          style={
            Object {
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
          x1="0"
          x2="100vw"
          y1="25"
          y2="25"
        />
      </svg>
      <svg
        height="50"
        width="50"
      >
        <polygon
          points="50,25 0,0 0,50"
          style={
            Object {
              "fill": "grey",
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
        />
      </svg>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "80vh",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100vw",
        }
      }
    >
      <svg
        height="50"
        width="50"
      >
        <polygon
          points="0,25 50,0 50,50"
          style={
            Object {
              "fill": "grey",
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
        />
      </svg>
      <svg
        height="50"
        width="100vw"
      >
        <line
          style={
            Object {
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
          x1="0"
          x2="100vw"
          y1="25"
          y2="25"
        />
      </svg>
      <svg
        height="50"
        width="50"
      >
        <polygon
          points="50,25 0,0 0,50"
          style={
            Object {
              "fill": "grey",
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
        />
      </svg>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "80vh",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100vw",
        }
      }
    >
      <svg
        height="50"
        width="50"
      >
        <polygon
          points="0,25 50,0 50,50"
          style={
            Object {
              "fill": "grey",
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
        />
      </svg>
      <svg
        height="50"
        width="100vw"
      >
        <line
          style={
            Object {
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
          x1="0"
          x2="100vw"
          y1="25"
          y2="25"
        />
      </svg>
      <svg
        height="50"
        width="50"
      >
        <polygon
          points="50,25 0,0 0,50"
          style={
            Object {
              "fill": "grey",
              "stroke": "red",
              "strokeWidth": 3,
            }
          }
        />
      </svg>
    </div>
  </div>
</div>
`;
